### PR TITLE
[82,84] - Create lot/auction (Save as draft)(Frontend)

### PR DESCRIPTION
**Pull request by otash8110**
### Where

[NL-82 Create lot/auction](https://jjbcapital.atlassian.net/browse/NL-82)

### What

Implement create-lot page for sellers and additional changes.

### How

add seller-navigation which is shown only to sellers. refactor authorize services and interceptors, configure routing. Add endpoint for receiving currencies metadata. Make pages addaptive

create lot endpoint: /seller/create-lot
seller dashboard endpoint: /seller/dashboard

User has to be logged in as a seller

### Affected Area

Auctionify.API.ClientApp
Auctionify.Application.Features.Currencies

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

- [x] 46 Tests Passed, 0 Ignored, 0 Failed

### CODE REVIEW

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
- Fix the responsiveness of the layout:
image
- Make it possible to add up to 20 photos
- According to the requirements, the maximum number of documents that can be uploaded is 3
- Validation errors should be displayed correctly:
image
- When you press Submit or Save as draft, both buttons are pressed at the same time
- Popups are not hidden when you click outside (consider replacing them with material ui built-in Dialog component)
- Fix this button
image

**Comment by khasanrashidov** [comment]
You have two User controllers: UsersController.cs and UserController.cs. Merge their contents and leave one as UsersController.cs.

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
- Remove all console.log() calls
- Buttons remains red after filling popups
image
image
- Fix the error message so that it is fully visible
image
- Could you please update these buttons so that after setting values in popups, they show those values as it is on figma?
image

File: src/Auctionify.API/ClientApp/src/app/components/seller/create-lot/create-lot.component.scss
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/seller/create-lot/create-lot.component.scss:1]: Remove empty rulesets

File: src/Auctionify.API/ClientApp/src/app/components/seller/create-lot/create-lot.component.html [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                                    <input formControlName="endDate" matInput type="datetime-local">
+                                    <mat-error *ngIf="lotForm.controls.endDate.hasError('required')">
+                                        End date is
+                                        <strong>required</strong>
+                                    </mat-error>
+                                </mat-form-field>
+                            </div>
+                        </div>
+                        <div class="lot-info__selector-inputs">
+                            <div class="buttons location-btn">
```
Comment (TsvirkoNikita) [inline src/Auctionify.API/ClientApp/src/app/components/seller/create-lot/create-lot.component.html:85]: What the reason to use classes without style (location-btn, starting-price-btn, file-btn)?

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

**Review by TsvirkoNikita — APPROVED** [review]

**Review by SaidganiUsm — APPROVED** [review]
Checked

**Pull request merged**